extensions/amp-iframely/validator-amp-iframely.protoascii

```diff
@@ -0,0 +1,65 @@
+#
+# Copyright 2021 The AMP HTML Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the license.
+#
+
+tags: {  # amp-iframely
+  html_format: AMP
+  tag_name: "SCRIPT"
+  extension_spec: {
+    name: "amp-iframely"
+    version: "0.1"
+    version: "latest"
+  }
+  attr_lists: "common-extension-attrs"
+}
+tags: {  # <amp-iframely>
+  html_format: AMP
+  tag_name: "AMP-IFRAMELY"
+  requires_extension: "amp-iframely"
+  # either data-id
+  attrs: {  # iframely generated ID of a widget
+    name: "data-id"
+    mandatory_oneof: "['data-id', 'data-url']"
+  }
+  attrs: { # iframe url to build widget from
+    name: "data-url"
+    mandatory_oneof: "['data-id', 'data-url']"
+  }
+  attrs: { name: "data-key" } # iframely key hash
```

Comment:
should this be mandatory? the comment below of `# optional attributes` makes me wonder if it is. if so then we need to add `mandatory: true` to `data-key`

Reply:
There are two ways to to identify content: 
1) either via ID generated by Iframely (and that would already link URL and key) 
2) or, via both url and key. 
So in that sense, `data-key` isn't required. But oneof id or URL is.

Reply:
We could have `data-url` also require `data-key` via a [trigger](https://github.com/ampproject/amphtml/blob/main/validator/validator.proto#L20).

[amp-twitter component](https://github.com/ampproject/amphtml/blob/0320666d03db37c4e0e363b55f828aa807ae321c/extensions/amp-twitter/validator-amp-twitter.protoascii#L33) uses it

For your case it'd look like this:

```
attrs: { # iframe url to build widget from
  name: "data-url"
  mandatory_oneof: "['data-id', 'data-url']"
  trigger: {
    also_requires_attr: "data-key"
  }
}
```

The required either of `data-id` and `data-url` would still be preserved.

Reply:
`also_requires_attr` trigger would indeed be really great, thanks. That is what we needed and I guess didn't notice it initially.